### PR TITLE
Feat/#9/bright event views

### PR DESCRIPTION
**Pull request by etormasa**
## Quick Info
Add views

## What does this change?
Added the view for the bright events, design for the view.
Modify routes 

## Why are you changing that?
the routes were modified for syntax error: sources -> resources

## How do you manually test this?
Using PUMA local server

## Screenshots 
image

### CODE REVIEW

File: app/assets/stylesheets/abstracts/_variables.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+$color-fuente-cabecera: #fff;
+$color-borde: #ddd;
+$color-fila-alterna: #f2f2f2;
+$color-hover: #ddd;
+
+
+$breakpoints-sm: 799px;
+$breakpoints-md: 800px;
+$breakpoints-lg: 1000px;
+$breakpoints-xl: 1200px;
```
Comment (JuanequeX) [inline app/assets/stylesheets/abstracts/_variables.scss]: set these variables in `_mixins.scss` file to manage in an easy way these breakpoints.

File: app/assets/stylesheets/components/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  color: black;
+  display: flex;
+  height: 50px;
+  justify-content: center;
+  margin: 2rem;
+  padding: 0;
+  text-align: center;
+  text-decoration: none;
+  transition: all 250ms ease-in-out;
+  width: 150px !important;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: try to avoid using `!important`

File: app/assets/stylesheets/components/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    transition: display 0.1s ease-in-out;
+    font-size: x-large;
+  }
+}
+
+@include desktop-lg {
+  .btn {
+    width: 15%;
+  }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_buttons.scss]: Add a blank line at the end of this file

File: app/assets/stylesheets/components/_hero.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+  h1 {
+    text-align: center;
+  }
+
+  img {
+    height: 350px;
+    width: auto;
+  }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_hero.scss]: add a blank line at the end of this file too.

File: app/assets/stylesheets/layout/_footer.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  background-color: $blue;
+  display: flex;
+  justify-content: center;
+  margin-top: 2rem;
+  padding: 1rem;
+
+  p {
+    color: $white;
+  }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_footer.scss]: add a blank line at the end of this file too.

File: app/assets/stylesheets/layout/_header.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    color: $white;
+    font-size: xx-large;
+  }
+}
+
+@include desktop-lg {
+  header i {
+    display: none;
+  }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/layout/_header.scss:24]: add a blank line at the end of this file too.

File: app/views/bright_events/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,50 @@
+<div>
+  <div>
+  <%= link_to bright_events_path , class: 'btn btn-add' do %>
+    <i class="uil uil-plus"></i>
+    <span>Agregar evento</span>
+  <% end %>
```
Comment (JuanequeX) [inline app/views/bright_events/index.html.erb]: set correctly this indentation to 2 spaces or 1 tab.

File: app/views/bright_events/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    <table>
+      <tr>
+        <th>Título</th>
+        <th>Descripción</th>
+        <th>Fecha</th>
+        <th>Ubicación</th>
+        <th>Editar</th>
+        <th>Ver</th>
+        <th>Eliminar</th>
+      </tr>
```
Comment (JuanequeX) [inline app/views/bright_events/index.html.erb]: if you're using local storage data through i18n, you can retrieve this kind of data from an es.yml file like this:

```
es:
  titles:
    title: 'Título'
    description: 'Descripción'
    date: 'Fecha'
    location: 'Ubicación'
    edit: 'Editar'
    view: 'Ver'
    delete: 'Eliminar'
```

With this approach, you will now be able to do something like this in your views:
```
 <tr>
    <% t('.titles').each do |key, value| %>
      <th><%= value %></th>
    <% end %>
  </tr>
```

File: app/views/bright_events/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+            <i class="uil uil-eye"></i>
+          </td>
+          <td>
+            <i class="uil uil-trash-alt"></i>
+          </td>
+        <% end %>      
+      </tr>
+    </table>
+  </div>
+</div>
```
Comment (JuanequeX) [inline app/views/bright_events/index.html.erb]: add a blank line at the end of this file too.

File: app/views/home/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<div class="hero">
+  <h1>Sistema de Gestion de Eventos</h1>
+  <%= image_tag('defaults/hero.jpg') %>
+
+  <%= link_to "Iniciar", bright_events_path , class: 'btn' %>
+</div>
```
Comment (JuanequeX) [inline app/views/home/index.html.erb]: add a blank line at the end of this file too.

File: app/views/layouts/_footer.html.erb [hunk 11]
```diff
@@ -0,0 +1,3 @@
+<footer>
+  <p>© BrightCoders Equipo 35. Todos los derechos reservados.</p>
+</footer>
```
Comment (JuanequeX) [inline app/views/layouts/_footer.html.erb:3]: add a blank line at the end of this file too.

File: app/views/layouts/_header.html.erb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+<header>
+    <div class="logo">
+        <%= image_tag('defaults/logo.png') %>
+    </div>
+    <i class="uil uil-bars"></i>
```
Comment (JuanequeX) [inline app/views/layouts/_header.html.erb]: indented correctly this code.

File: config/routes.rb [hunk 13]
```diff
@@ -6,5 +6,9 @@
   get "up" => "rails/health#show", as: :rails_health_check
 
   # Defines the root path route ("/")
-  # root "posts#index"
+  root 'home#index'
+
+  get '/home' => 'home#index'
```
Comment (JuanequeX) [inline config/routes.rb]: what is the reason to use the same content in `root` and `/home`

**Review by JuanequeX — DISMISSED** [review]

File: app/controllers/bright_events_controller.rb [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+  def index
+    @events = BrightEvent.all
+  end
+
+  def show
+    @event = BrightEvent.find(params[:id])
+  end
+
+  def new
+    @events = BrightEvent.new
```
Comment (MauricioNC) [inline app/controllers/bright_events_controller.rb:11]: This is ok, but you are repeating this line many times, instead you can use the before_action filter.

See the documentation at:
https://guides.rubyonrails.org/action_controller_overview.html#filters

File: app/controllers/bright_events_controller.rb [hunk 15]
```diff
@@ -0,0 +1,52 @@
+class BrightEventsController < ApplicationController
+  def index
+    @events = BrightEvent.all
+  end
+
+  def show
+    @event = BrightEvent.find(params[:id])
+  end
```
Comment (MauricioNC) [inline app/controllers/bright_events_controller.rb]: This is ok, but you are repeating this line many times, instead you can use the before_action filter.

See the documentation at:
https://guides.rubyonrails.org/action_controller_overview.html#filters

**Comment by JuanequeX** [comment]
Another recommendation that I forgot to specify in the last comments is to try to avoid using images in the `.jpg` format and instead change them to `.webp`. Believe me, this format provides better performance during the asset precompilation process.

File: test/controllers/bright_events_controller_test.rb [hunk 16]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class BrightEventsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (juperameza) [inline test/controllers/bright_events_controller_test.rb]: If you dont need this delete it

File: test/controllers/home_controller_test.rb [hunk 17]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class HomeControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (juperameza) [inline test/controllers/home_controller_test.rb]: Same here

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/abstracts/_variables.scss (hunk 1 above)
Comment (etormasa) [reply app/assets/stylesheets/abstracts/_variables.scss]: done  :+1:

File: app/assets/stylesheets/components/_buttons.scss (hunk 3 above)
Comment (etormasa) [reply app/assets/stylesheets/components/_buttons.scss]: done  :+1:

File: app/assets/stylesheets/layout/_footer.scss (hunk 5 above)
Comment (etormasa) [reply app/assets/stylesheets/layout/_footer.scss]: done  :+1:

File: app/views/bright_events/index.html.erb (hunk 9 above)
Comment (etormasa) [reply app/views/bright_events/index.html.erb]: done  :+1:

File: app/views/layouts/_footer.html.erb (hunk 11 above)
Comment (etormasa) [reply app/views/layouts/_footer.html.erb:3]: done  :+1:

File: config/routes.rb (hunk 13 above)
Comment (etormasa) [reply config/routes.rb]: done  :+1:

File: app/controllers/bright_events_controller.rb (hunk 15 above)
Comment (etormasa) [reply app/controllers/bright_events_controller.rb]: done  :+1:

File: test/controllers/bright_events_controller_test.rb (hunk 16 above)
Comment (etormasa) [reply test/controllers/bright_events_controller_test.rb]: done  :+1:

File: test/controllers/home_controller_test.rb (hunk 17 above)
Comment (etormasa) [reply test/controllers/home_controller_test.rb]: done  :+1:

File: app/assets/stylesheets/components/_hero.scss (hunk 4 above)
Comment (etormasa) [reply app/assets/stylesheets/components/_hero.scss]: done :+1:

File: app/assets/stylesheets/components/_buttons.scss (hunk 2 above)
Comment (etormasa) [reply app/assets/stylesheets/components/_buttons.scss]: done :+1:

File: app/views/layouts/_header.html.erb (hunk 12 above)
Comment (etormasa) [reply app/views/layouts/_header.html.erb]: done :+1:

**Review by juperameza — APPROVED** [review]
LGTM!

File: app/controllers/bright_events_controller.rb [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+  private
+
+  def event_params
+    params.require(:bright_event).permit(:title, :description, :date, :ubication, :cost)
+  end
+
+  def find_event
+    BrightEvent.find(params[:id])
+  end
+end
```
Comment (MauricioNC) [inline app/controllers/bright_events_controller.rb:58]: You don't need to call the find_event method. It is called and executed before any action for :show, :edit, :update and :destroy

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/controllers/bright_events_controller.rb [hunk 19]
```diff
@@ -0,0 +1,58 @@
+class BrightEventsController < ApplicationController
+  before_action :find_event, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @events = BrightEvent.all
+  end
+
+  def show
+    @event = find_event
+  end
```
Comment (MauricioNC) [inline app/controllers/bright_events_controller.rb]: You don't need to call the find_event method. It is called and executed before any action for :show, :edit, :update and :destroy
Delete this line from the other methods too.

In the find_event you can write it as:

@ event = BrightEvent.find(params[:id])

Then you can call only the @ event in every method if needed

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/controllers/bright_events_controller.rb (hunk 19 above)
Comment (etormasa) [reply app/controllers/bright_events_controller.rb]: Done :+1:

**Review by MauricioNC — APPROVED** [review]

**Pull request merged**